tests/ui/clear_with_drain.rs

```diff
@@ -0,0 +1,84 @@
+// run-rustfix
+#![allow(unused)]
+#![warn(clippy::clear_with_drain)]
+
+fn range() {
+    let mut v = vec![1, 2, 3];
+    let iter = v.drain(0..v.len()); // Yay
+
+    let mut v = vec![1, 2, 3];
+    let n = v.drain(0..v.len()).count(); // Yay
+
+    let mut v = vec![1, 2, 3];
+    let n = v.drain(usize::MIN..v.len()).count(); // Yay
+
+    let mut v = vec![1, 2, 3];
+    v.drain(0..v.len()); // Nay
+
+    let mut v = vec![1, 2, 3];
+    v.drain(usize::MIN..v.len()); // Nay
+}
+
+fn range_from() {
+    let mut v = vec![1, 2, 3];
+    let iter = v.drain(0..); // Yay
+
+    let mut v = vec![1, 2, 3];
+    let next = v.drain(0..).next(); // Yay
```

Comment:
Could we have an example where the iter is only used later?

Reply:
Sure thing!